workflow/sync/semaphore.go

```diff
@@ -175,11 +186,10 @@ func (s *PrioritySemaphore) tryAcquire(holderKey string) (bool, string) {
 	// If it is not a front key, it needs to wait for its turn.
 	if s.pending.Len() > 0 {
 		item := s.pending.peek()
-		nextKey = fmt.Sprintf("%v", item.key)
-		if holderKey != nextKey && !isSameWorkflowNodeKeys(holderKey, nextKey) {
+		if holderKey != nextKey && !isSameWorkflowNodeKeys(holderKey, item.key) {
 			// Enqueue the front workflow if lock is available
 			if len(s.lockHolder) < s.limit {
-				s.nextWorkflow(nextKey)
```

Comment:
this appears to be a bug with notify implementation. nextKey might be in the form of `namespace/workflowname/nodeid`, but we must enqueue it as `namespace/workflowname`